docs/source/wrappers.rst

```diff
@@ -0,0 +1,499 @@
+offer an easy way to compile and tune compilation parameters. Examples:
+
+.. code:: python
+
+    def model_build_fn(compile_kwargs):
+        # you can access the ``optimizer`` param here
+        optimizer = compile_kwargs["optimizer"]
+        if optimizer is None:
+            # and apply any custom logic you wish
+            ...
+        model = Model(...)
+        ...
+        model.compile(optimizer=optimizer)
+        return model
+
+    clf = KerasClassifier(model=model_build_fn)
+    clf.fit(X, y)
+    y_pred = clf.predict(X_valid)
+
+.. code:: python
+
+    from tensorflow.keras.optimizers import Adam
+
+    def model_build_fn():
+        model = Model(...)
+        ...
+        # Do not call model.compile
+        return model  # That's it, SciKeras will compile your model for you
+
+    clf = KerasClassifier(model=model_build_fn, optimizer=Adam)
+    clf.fit(X, y)
+    y_pred = clf.predict(X_valid)
+
+
+In all cases, returning an un-compiled model is equivalent to
+calling ``model.compile(**compile_kwargs)`` within ``model_build_fn``.
+
+
+Arguments to ``model_build_fn``
+-------------------------------
+
+User-defined keyword arguments passed to :py:func:`~scikeras.wrappers.BaseWrapper.__init__`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+All keyword arguments that were given to :py:func:`~scikeras.wrappers.BaseWrapper.__init__`
+will be passed to ``model_build_fn`` directly.
+For example, calling ``KerasClassifier(myparam=10)`` will result in a
+``model_build_fn(my_param=10)`` call.
+Note however that ``KerasClassifier(optimizer="sgd")`` will **not** result in
+``model_build_fn(optimizer="sgd")``. Instead, you must access ``optimizer`` either
+via ``compile_kwargs`` if you want a compiled optimizer
+or ``params`` if you want the raw input.
+
+Optional arguments
+^^^^^^^^^^^^^^^^^^
+
+You may want to use attributes from
+:py:class:`~scikeras.wrappers.BaseWrapper` such as ``n_features_in_`` while building
+your model, or you may wish to let SciKeras compile your optimizers and losses
+but apply some custom logic on top of that compilation.
+
+To enable this, SciKeras uses three special arguments to ``model`` that will only
+be passed if they are present in ``model``'s signature (i.e. there is an argument
+with the same name in ``model``'s signature):
+
+``meta``
+++++++++
+This is a dictionary containing all of the attributes that
+:py:class:`~scikeras.wrappers.BaseWrapper` creates when it is initialized
+These include ``n_features_in_``, ``y_dtype_``, etc. For a full list,
+see the :ref:`scikeras-api` documentation.
+
+``compile_kwargs``
+++++++++++++++++++++++++
+This is a dictionary of parameters destined for :py:func:`tensorflow.Keras.Model.compile`.
+This dictionary can be used like ``model.compile(**compile_kwargs)``.
+All optimizers, losses and metrics will be compiled if classes with routed parameters
```

Comment:
Doesn't passing `optimizer="sgd"` always result in `instance(compile_kwargs["optimizer"], tf.keras.optimizers.SGD`?

Reply:
Yep you're right, that's outdated.

Reply:
Changed to:

```rst
All optimizers, losses and metrics will be compiled to objects,
even if string shorthands (e.g. ``optimizer="adam"``) were passed.
```